### ISSUE TITLE
Null reference exception on startup when no assets

### ISSUE THREAD

**Issue by appurist** [HalcyonGrid/halcyon#21]
Started a fresh server up tonight with a fresh WHIP installation (no assets) and the Halcyon region server crashed at startup generating the map tile (presumably when it couldn't fetch an image asset).  This may also be related to the fact that I had Cloud Files disabled on this startup as well. (I haven't investigated yet but I thought I'd make sure this was tracked.)

```
Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at InWorldz.Data.Assets.Stratus.StratusAssetClient.TryRequestAsset(UUID assetID, IAssetServer server, AssetBase& asset) in C:\DevSL\InWorldz\halcyon\InWorldz\InWorldz.Data.Assets.Stratus\StratusAssetClient.cs:line 237
   at InWorldz.Data.Assets.Stratus.StratusAssetClient.RequestAssetSync(UUID assetID) in C:\DevSL\InWorldz\halcyon\InWorldz\InWorldz.Data.Assets.Stratus\StratusAssetClient.cs:line 221
   at OpenSim.Framework.Communications.Cache.AssetCache.GetAsset(UUID assetID, AssetRequestInfo reqInfo) in C:\DevSL\InWorldz\halcyon\OpenSim\Framework\Communications\Cache\AssetCache.cs:line 267
   at OpenSim.Region.CoreModules.World.WorldMap.TexturedMapTileRenderer.fetchTexture(UUID id) in C:\DevSL\InWorldz\halcyon\OpenSim\Region\CoreModules\World\WorldMap\TexturedMapTileRenderer.cs:line 173
   at OpenSim.Region.CoreModules.World.WorldMap.TexturedMapTileRenderer.computeAverageColor(UUID textureID, Color defaultColor) in C:\DevSL\InWorldz\halcyon\OpenSim\Region\CoreModules\World\WorldMap\TexturedMapTileRenderer.cs:line 234
   at OpenSim.Region.CoreModules.World.WorldMap.TexturedMapTileRenderer.TerrainToBitmap(DirectBitmap mapbmp) in C:\DevSL\InWorldz\halcyon\OpenSim\Region\CoreModules\World\WorldMap\TexturedMapTileRenderer.cs:line 288
   at OpenSim.Region.CoreModules.World.WorldMap.MapImageModule.WriteJpeg2000Image() in C:\DevSL\InWorldz\halcyon\OpenSim\Region\CoreModules\World\WorldMap\MapImageModule.cs:line 122
   at OpenSim.Region.Framework.Scenes.Scene.CreateTerrainTexture(Boolean temporary) in C:\DevSL\InWorldz\halcyon\OpenSim\Region\Framework\Scenes\Scene.cs:line 1932
   at OpenSim.OpenSimBase.CreateRegion(RegionInfo regionInfo, Boolean portadd_flag, Boolean do_post_init, IScene& mscene) in C:\DevSL\InWorldz\halcyon\OpenSim\Base\OpenSimBase.cs:line 474
   at OpenSim.OpenSimBase.CreateRegion(RegionInfo regionInfo, Boolean portadd_flag, IScene& scene) in C:\DevSL\InWorldz\halcyon\OpenSim\Base\OpenSimBase.cs:line 395
   at OpenSim.ApplicationPlugins.LoadRegions.LoadRegionsPlugin.PostInitialize() in C:\DevSL\InWorldz\halcyon\OpenSim\ApplicationPlugins\LoadRegions\LoadRegionsPlugin.cs:line 119
   at OpenSim.OpenSimBase.StartupSpecific() in C:\DevSL\InWorldz\halcyon\OpenSim\Base\OpenSimBase.cs:line 211
   at OpenSim.OpenSim.StartupSpecific() in C:\DevSL\InWorldz\halcyon\OpenSim\Base\OpenSim.cs:line 163
   at OpenSim.Framework.Servers.BaseOpenSimServer.Startup() in C:\DevSL\InWorldz\halcyon\OpenSim\Framework\Servers\BaseOpenSimServer.cs:line 300
   at OpenSim.Application.Main(String[] args) in C:\DevSL\InWorldz\halcyon\InWorldz\Halcyon\Application.cs:line 153
```

**Comment by kf6kjg** [comment]
The only line that could have NRE'd in there means that somehow the asset server was disabled.  Seeing as you mentioned disabling CF, dis you remember to leave `LegacySupport` true? Having both `LegacySupport` and CF being false means no asset server at all.

**Comment by appurist** [comment]
LegacySupport was true.  The config wasn't the problem, it was just a logic problem in the code where it always fell back to the second server if the asset wasn't found (see simple code in image below) even if the second server was null.  I needed this resolved (blocking problem) so I committed a fix (but oops in the master branch).  So it's resolved now in master.  I'm not sure why this wasn't more common, since it happens if Cloud Files is not in use, which is probably true on almost every other Halcyon server.

```
[InWorldz.Data.Assets.Stratus]
 CFSupport = false
 LegacySupport = true
 WriteTarget = "whip"
 WhipURL = "whip://password@host:32700"
```
![fallback code](https://i.gyazo.com/347f2f2638111e3b80bfbfed510a7eeb.png)

**Comment by appurist** [comment]
This was resolved by commit 8b0afefc578a25cd80311bedae1c1d72d6ec965e
